docs/contributing/contributing.md

```diff
@@ -221,7 +221,7 @@ Run the benchmarks with `cabal bench`.
 
 It should take around 25 minutes and the results will be stored in the `bench-results` folder. To interpret the results, see the comments in the `bench/Main.hs` module.
 
-More details in [bench/README](../../bench/README.md)
+More details in [bench/README](https://github.com/haskell/haskell-language-server/blob/master/bench/README.md)
```

Comment:
These links are referring to stuff outside of docs directory, so sphinx is unable to create proper links.
Rather than trying to change sphinx configuration, i think it's simpler to just link to relevant docs on master branch..?

Reply:
will it cause a version mismatch though🤔

Reply:
Let's think through the implications of this change:

1) Before this PR, the links (not images, which are copied during the build!) that point to relative file paths outside of docs directory are not rendered as links:
Example: search for `bench/README` on [this page](https://haskell-language-server.readthedocs.io/en/stable/contributing/contributing.html).

When building the docs by running `make html` in the docs folder you get the following warning:
/home/jhrcek/Devel/github.com/haskell/haskell-language-server/docs/contributing/contributing.md:224: WARNING: Unknown source document '/home/jhrcek/Devel/github.com/haskell/haskell-language-server/bench/README' [myst.xref_missing]

And the resulting html has no links:
![Screenshot from 2024-01-30 05-34-52](https://github.com/haskell/haskell-language-server/assets/2716069/9f8f300c-2e62-425c-a001-bb121b43a39e)

2) With this PR: the link is renered correctly and points to readme on master:
![Screenshot from 2024-01-30 05-40-02](https://github.com/haskell/haskell-language-server/assets/2716069/c909ff4e-816e-4d25-a07f-24edd9ae31b8)

You're right that this is not ideal, because the readme on master can be modified in a way that makes is "incompatible" with the rendered version of the docs or it can be even removed etc. But this is what most of the other links on the page are currently doing (just pointing to master version of the doc on github).

3)The ideal solution would be to replace "master" within those links with "the commit hash of the commit from which the docs were built" which would tie the rendered docs to the state of the repo when they were built. Unfortunately I don't know what's the "canonical" way to do this with sphinx (I can only think of hacks like adding a placeholder in all those links and running `sed` before building with shinx, which feels like hack).